cub/cub/thread/thread_operators.cuh

```diff
@@ -374,93 +375,74 @@ struct ReduceByKeyOp
   }
 };
 
 template <>
-struct SimdMin<::cuda::std::uint16_t>
+struct SimdMin<uint16_t>
 {
-  using simd_type = ::cuda::std::uint32_t;
-
-  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE ::cuda::std::uint32_t
-  operator()(::cuda::std::uint32_t a, ::cuda::std::uint32_t b) const
+  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE uint32_t operator()(uint32_t a, uint32_t b) const
   {
-    return __vminu2(a, b);
+    NV_IF_TARGET(NV_PROVIDES_SM_90,
+                 (return __vminu2(a, b);), //
+                 (return simd_operation_is_not_supported_before_sm90();));
   }
 };
 
 #  if _CCCL_HAS_NVFP16()
 
+_CCCL_HOST_DEVICE __half2 simd_operation_is_not_supported_before_sm80(__half2);
+
 template <>
 struct SimdMin<__half>
 {
-  using simd_type = __half2;
-
-  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __half2 operator()(__half2 a, __half2 b) const
+  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __half2
+  operator()([[maybe_unused]] __half2 a, [[maybe_unused]] __half2 b) const
   {
-#    if _CCCL_CUDACC_BELOW(12) && _CCCL_CUDA_COMPILER(NVHPC)
-    return __floats2half2_rn(::cuda::minimum<>{}(__half2float(a.x), __half2float(b.x)),
-                             ::cuda::minimum<>{}(__half2float(a.y), __half2float(b.y)));
-#    else // ^^^ _CCCL_CUDACC_BELOW(12) && _CCCL_CUDA_COMPILER(NVHPC) ^^^ / vvv otherwise vvv
     NV_IF_TARGET(NV_PROVIDES_SM_80,
-                 (return __hmin2(a, b);),
-                 (return __halves2half2(__float2half(::cuda::minimum<>{}(__half2float(a.x), __half2float(b.x))),
-                                        __float2half(::cuda::minimum<>{}(__half2float(a.y), __half2float(b.y))));));
-#    endif // !_CCCL_CUDACC_BELOW(12) || !_CCCL_CUDA_COMPILER(NVHPC)
+                 (return __hmin2(a, b);), //
+                 (return simd_operation_is_not_supported_before_sm80(__half2{});));
   }
 };
 
 #  endif // _CCCL_HAS_NVFP16()
 
 #  if _CCCL_HAS_NVBF16()
 
-// NOTE: __halves2bfloat162 is not always available on older CUDA Toolkits for __CUDA_ARCH__ < 800
-_CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __nv_bfloat162 halves2bfloat162(__nv_bfloat16 a, __nv_bfloat16 b)
-{
-  ::cuda::std::uint32_t tmp;
-  auto a_uint16 = ::cuda::std::bit_cast<::cuda::std::uint16_t>(a);
-  auto b_uint16 = ::cuda::std::bit_cast<::cuda::std::uint16_t>(b);
-  asm("{mov.b32 %0, {%1,%2};}\n" : "=r"(tmp) : "h"(a_uint16), "h"(b_uint16));
-  __nv_bfloat162 ret;
-  ::memcpy(&ret, &tmp, sizeof(ret));
-  return ret; // TODO: replace with ::cuda::std::bit_cast<__nv_bfloat162>(tmp);
-}
+_CCCL_HOST_DEVICE __nv_bfloat162 simd_operation_is_not_supported_before_sm80(__nv_bfloat162);
 
 template <>
 struct SimdMin<__nv_bfloat16>
 {
-  using simd_type = __nv_bfloat162;
-
-  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __nv_bfloat162 operator()(__nv_bfloat162 a, __nv_bfloat162 b) const
+  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __nv_bfloat162
+  operator()([[maybe_unused]] __nv_bfloat162 a, [[maybe_unused]] __nv_bfloat162 b) const
```

Comment:
Nitpick: We are inconsistent about the use of `[[maybe_unused]]` in this file. Can we either drop it or use it everywhere?